experiment/simulation/js/animation-ring-counter.js

```diff
@@ -0,0 +1,567 @@
+"use strict";
+// Dimensions of working area
+const circuitBoard = document.getElementById("circuit-board");
+const sidePanels = document.getElementsByClassName("v-datalist-container");
+// Distance of working area from top
+const circuitBoardTop = circuitBoard.offsetTop;
+// Full height of window
+
+const windowHeight = window.innerHeight;
+const width = window.innerWidth;
+const instructionBox = document.getElementsByClassName("instructions-box")[0];
+
+const svg = document.querySelector(".svg");
+const inputpath1 = document.querySelector("#inputpath1");
+const svgns = "http://www.w3.org/2000/svg";
+
+
+// stroing the necessary div elements in const
+const STATUS = document.getElementById("playOrPause");
+const OBSERV = document.getElementById("Observations");
+const SPEED = document.getElementById("speed");
+
+// global varaibles declared here
+const OBJECTS = [document.getElementById("ori"),document.getElementById("clock"),document.getElementById("qa"),document.getElementById("qb"),document.getElementById("qc")];
+const TEXTINPUT = [document.createElementNS(svgns, "text")];
+const TEXTCLOCK = [document.createElementNS(svgns, "text")];
+const TEXTOUTPUT = [document.createElementNS(svgns, "text"),document.createElementNS(svgns, "text"),document.createElementNS(svgns, "text")];
+const ORIDOT = [document.createElementNS(svgns, "circle"),document.createElementNS(svgns, "circle"),document.createElementNS(svgns, "circle")];
+const CLOCKDOT = [document.createElementNS(svgns, "circle"),document.createElementNS(svgns, "circle"),document.createElementNS(svgns, "circle")];
+
+let timeline = gsap.timeline({ repeat: 0, repeatDelay: 0 });
+
+// decide help to decide the speed
+let decide = 0;
+// circuitStarted is initialised to 0 which depicts that demo hasn't started whereas circuitStarted 1 depicts that the demo has started.
+let circuitStarted = 0;
+
+
+// function to take care of width
+function demoWidth() {
+    if (width < 1024) {
+        circuitBoard.style.height = 600 + "px";
+    } else {
+        circuitBoard.style.height = windowHeight - circuitBoardTop - 20 + "px";
+    }
+    sidePanels[0].style.height = circuitBoard.style.height;
+}
+
+// function to initialise the instruction box
+function instructionBoxInit() {
+    // Instruction box
+    instructionBox.addEventListener("click", () => {
+        instructionBox.classList.toggle("expand");
+    });
+}
+
+function setCoordinates(xObject,yObject,textObject){
+    gsap.set(textObject,{
+        x: xObject,
+        y: yObject
+    })
+}
+
+function fillColor(object,color){
+    gsap.set(object, {
+        fill: color
+    });
+}
+
+function fillInputDots(object,cxObject,cyObject,rObject,fillObject) {
```

Comment:
A lot of these helper functions appear duplicated in both basic counter and ring counter files. Can we move them out into a separate utility file and pass in the gsap instance and share those functions in both files?

Reply:
Made a new file animation-utility that contains all these functions